lib/util/memcache.ex

```diff
@@ -4,12 +4,51 @@ use Croma
 
 defmodule Antikythera.Memcache do
   @moduledoc """
-  In-memory cache for each executor pool.
+  Easy-to-use in-memory cache for each executor pool.
 
-      iex> Antikythera.Memcache.write("foo", "bar", epool_id)
+  #{inspect(__MODULE__)} behaves as a key-value storage.
+  Cached key-value pairs are internally stored in ETS.
+  It accepts arbitrary (but not too large) terms as both keys and values.
+
+  ## Usage
+
+      iex> Antikythera.Memcache.write("foo", "bar", epool_id, 3_600)
       :ok
       iex> Antikythera.Memcache.read("foo", epool_id)
       {:ok, "bar"}
+
+  ## Limits
+
+  The number of records and the size of keys and values are limited.
+
+  - The maximum number of records for each executor pool is #{AntikytheraCore.ExecutorPool.MemcacheWriter.max_records()}.
+      - If exceeds the limit, a record nearest to expiration is evicted so that a new record can be inserted.
+  - The maximum size of keys and values is defined in `Antikythera.Memcache.Key` and `Antikythera.Memcache.Value`.
+      - To know how the size of keys and values is calculated, see `Antikythera.TermUtil`.
+      - If exceeds the limit, `write/5` returns an error `:too_large_key` or `:too_large_value`.
+
+  ## Lifetime of records
+
+  There are 2 cases where records in #{inspect(__MODULE__)} are evicted:
+
+  1. Records are expired (see [Mechanism of Expiration](#module-mechanism-of-expiration) below for more details)
+  2. Reach the maximum number of records for each executor pool
+
+  If it is the case of (2), records whose expiration time is nearest are evicted to keep the maximum number of records.
+
+  Please note that records in #{inspect(__MODULE__)} could be evicted anytime.
+
+  ## Mechanism of Expiration
+
+  The lifetime of records must be set as `lifetime_in_sec` in `write/5`.
+  This lifetime does not guarantee that records remain in the entire specified lifetime.
+
+  To avoid the thundering herd, whether records are expired is decided probabilistically.
```

Comment:
It would be kind to put a link to some resource about "thundering herd problem" since it is a technical term first mentioned here.
I could not find a very good one but a Wikipedia link should suffice as an entry point: https://en.wikipedia.org/wiki/Thundering_herd_problem

Reply:
Added a link in 089fa6487a6319087f67c28dcce9c115e879e845